packages/core/src/types.ts

```diff
@@ -1603,16 +1603,20 @@ export interface Subscription {
 }
 
 export interface InteropObservable<T> {
-  [Symbol.observable]: () => Subscribable<T>;
+  [Symbol.observable]: () => InteropSubscribable<T>;
 }
 
-export interface Subscribable<T> {
+export interface InteropSubscribable<T> {
+  subscribe(observer: Observer<T>): Subscription;
+}
```

Comment:
In a way, this breaks the previous contract exposed under `Symbol.observable` method. However, this is the only signature that is valid from Rx's PoV. The `InteropObservable` returns `Subscribable<T>`:
https://github.com/ReactiveX/rxjs/blob/67cb317a7a6b9fdbd3d2e8fdbc2ac9ac7e57179c/api_guard/dist/types/index.d.ts#L321-L323
and Subscribable is defined as follows:
https://github.com/ReactiveX/rxjs/blob/67cb317a7a6b9fdbd3d2e8fdbc2ac9ac7e57179c/api_guard/dist/types/index.d.ts#L690-L692

So, in fact, this interface only accepts observers - it doesn't accept separate listener functions for `next`, `error` and `complete`.

By implementing this change and using this as an explicit return type of the `Symol.observable` method on the `Interpreter` class we ensure that it's compatible with the Rx's signature for this:
https://github.com/statelyai/xstate/blob/d912a8d8c00d6dc00787235f31ae1344d8750269/packages/core/src/interpreter.ts#L1369-L1373

Thanks to that we can revert the reordering of the `subscribe`'s overloads from [this PR](https://github.com/statelyai/xstate/pull/2371/files) and fix the issue with Svelte's readables. This is because both libraries are using conditional types to match the input argument and to "extract" the emitted type and conditional types are only seeing the last overload.

So, in a sense, this is a hack and might break any minute... cause the whole thing is fitting both libraries thanks to the alignment of the stars and not because this is built on super solid principles. We trick Rx by only exposing a single overload with the observer as the parameter for its conditional type from the `Symbol.observable` method and we trick Svelte by reordering overloads so it only sees the one with callbacks parameters.